### PR TITLE
Fix some of the comments from Ocramius

### PR DESCRIPTION
**Pull request by eigan**
Just some of the [comments from Ocramius](https://github.com/doctrine/dbal/pull/2450) in your pull requests.

**I left out:**
date_create_immutable -> DateTimeImmutable.
The DateTimeImmutable constructor threw a message: `The timezone could not be found in the database`

See [my comment](https://github.com/doctrine/dbal/pull/2450#issuecomment-293825763) regarding these changes, and the rest of his comments.

### CODE REVIEW

File: tests/Doctrine/Tests/DBAL/Types/DateImmutableTypeTest.php [hunk 1]
```diff
@@ -57,11 +57,10 @@ public function testConvertsNullToDatabaseValue()
         $this->assertNull($this->type->convertToDatabaseValue(null, $this->platform->reveal()));
     }
 
-    /**
-     * @expectedException \Doctrine\DBAL\Types\ConversionException
-     */
     public function testDoesNotSupportMutableDateTimeToDatabaseValueConversion()
     {
+        $this->setExpectedException('Doctrine\DBAL\Types\ConversionException');
```
Comment (lcobucci) [inline tests/Doctrine/Tests/DBAL/Types/DateImmutableTypeTest.php]: It should be `$this->expectException()` and use `::class` here, right?

File: tests/Doctrine/Tests/DBAL/Types/DateImmutableTypeTest.php [hunk 2]
```diff
@@ -96,11 +95,10 @@ public function testResetTimeFractionsWhenConvertingToPHPValue()
         $this->assertSame('2016-01-01 00:00:00.000000', $date->format('Y-m-d H:i:s.u'));
     }
 
-    /**
-     * @expectedException \Doctrine\DBAL\Types\ConversionException
-     */
     public function testThrowsExceptionDuringConversionToPHPValueWithInvalidDateString()
     {
+        $this->setExpectedException('Doctrine\DBAL\Types\ConversionException');
```
Comment (lcobucci) [inline tests/Doctrine/Tests/DBAL/Types/DateImmutableTypeTest.php]: Same as https://github.com/deeky666/dbal/pull/5/files#r111340570

**Comment by eigan** [comment]
I just fixed up some more stuff I had missed. And squashed..

@lcobucci Should be fixed now

**Comment by deeky666** [comment]
@eigan oh sorry I didn't see you created this PR :(

**Comment by eigan** [comment]
@deeky666 heh, thats allright :)

**Pull request closed without merging**